static/js/settings_bots.js

```diff
@@ -17,6 +17,16 @@ function is_local_part(value, element) {
     return this.optional(element) || /^[\-!#$%&'*+\/=?\^_`{}|~0-9A-Z]+(\.[\-!#$%&'*+\/=?\^_`{}|~0-9A-Z]+)*$/i.test(value);
 }
 
+function type_id_to_string(type_id) {
+    if (type_id === 1) {
+        return "Generic bot";
+    } else if (type_id === 2) {
+        return "Incoming webhook";
+    } else if (type_id === 3) {
+        return "Outgoing webhook";
+    }
```

Comment:
I'd also deduplicate this list with the one in `static/templates/settings/bot-settings.handlebars` -- you can probably just pass a dictionary into that function as part of the template context.